### PR TITLE
[CI] OSX tester: change to gcc-10.

### PR DESCRIPTION
**Pull request by luca-heltai**
Try to fix osx build.

### CODE REVIEW

**Review by peterrum — APPROVED** [review]

File: .github/workflows/osx.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
         ./contrib/utilities/download_clang_format
     - name: info
       run: |
-        g++-9 -v
+        g++-10 -v
```
Comment (masterleinad) [inline .github/workflows/osx.yml:53]: Why do we print this at all if we don't use `gcc` here?

Comment (luca-heltai) [reply .github/workflows/osx.yml:53]: I think we do. for 64bits we use mpicxx (based on gcc). I'd be ok to remove it, but I think it is a good idea to keep in the logs what has been used.

**Comment by luca-heltai** [comment]
the other tests are not necessary I guess. If you want, I can remove the info stage (or make it based on mpicxx instead of gcc-XX).

**Comment by luca-heltai** [comment]
I'll merge. If you want me to make more changes, I'll do this in a followup PR.

**Pull request merged**